app/controllers/admin_controller.rb

```diff
@@ -123,6 +123,38 @@ def mark_spam
     end
   end
 
+  def mark_comment_spam
+    @comment = Comment.find params[:id]
+    if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
+      if @comment.status == 0 || @comment.status == 1
+        @comment.spam
+        flash[:notice] = "Comment has been marked as spam."
+      else
+        flash[:notice] = "Comment already marked as spam."
+      end
```

Comment:
I'd like to make one last suggestion. The idea is to first check if the status is as desired, in which case you move on, or else you set it as desired. I think it'll look as follows:
```
if status is as desired
  flash the "already" msg
else
  set status as desired
  flash msg about what you did there
end
```
This is what you did in the publish_comment method, and it is clearer IMO.